indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,17 +125,66 @@ public IndexTask(
       @JsonProperty("context") final Map<String, Object> context
   )
   {
-    super(
-        makeId(id, ingestionSchema),
-        makeGroupId(ingestionSchema),
+    this(
+        id,
         taskResource,
-        makeDataSource(ingestionSchema),
+        ingestionSchema.dataSchema.getDataSource(),
+        ingestionSchema.ioConfig.appendToExisting,
+        ingestionSchema,
+        context
+    );
+  }
+
+  IndexTask(
+      String id,
+      TaskResource resource,
+      String dataSource,
+      boolean appendToExisting,
+      IndexIngestionSpec ingestionSchema,
+      Map<String, Object> context
+  )
+  {
+    this(
+        getOrMakeId(id, TYPE, dataSource),
+        makeGroupId(appendToExisting, dataSource),
+        resource,
+        dataSource,
+        ingestionSchema,
+        context
+    );
+  }
+
+  IndexTask(
```

Comment:
This is really a lot of constructors. Maybe a builder pattern would be better. (It can be difficult when writing and reading code to understand what constructor is used)

Reply:
Removed one of constructors.